Game controller handlers

MettyS opened the pull request.

MettyS committed:
Add internal message handling

Create game_handler.go to represent a game instance.
Basic game creation logic and initializers.
Add shared package and utils.go to hold internal grpc
translation structs.

Add playerID and gameID fields to some messages in:
game_control.proto
game.proto

jer-irl commented on server/controller/central_checkers_controller.go:
```diff
@@ -2,95 +2,108 @@ package controller
+// d.GameStartRequest expected fields:
+// GameID string - optional
+func validateGameStartRequest(req d.GameStartRequest) (bool, string) {
+	return true, ""
```
Can you add a TODO comment here?

jer-irl commented on server/controller/central_checkers_controller.go:
```diff
@@ -2,95 +2,108 @@ package controller
+	if gameInstance, exists := GameList[gameID]; exists {
+		playerRole, err = gameInstance.AddParticipant(playerID)
+	} else {
+		gameID = "RandomGameIDOverwriteChangeThis"
```
TODO comment, please!

jer-irl commented on server/controller/central_checkers_controller.go:
```diff
@@ -2,95 +2,108 @@ package controller
-	return GameStartResponse{}, nil
+func HandleStartGame(req d.GameStartRequest) (d.GameStartResponse, error) {
+	gameID := req.GameID
+	playerID := "RandomKeyChangeThis"
```
TODO comment here, please!

jer-irl commented on server/services/gamecontrol.go:
```diff
@@ -12,19 +13,20 @@ type GameControlServer struct {
-func convertGameStartRequestInward(req *pb.GameStartRequest) controller.GameStartRequest {
-	domGameStartRequest := controller.GameStartRequest{}
+func convertGameStartRequestInward(req *pb.GameStartRequest) d.GameStartRequest {
+	domGameStartRequest := d.GameStartRequest{}
```
I personally prefer a name that is harder to confuse with "DOM."  Like `domain` as a prefix instead.

MettyS replied:
I refactored this prefix on this advice.

MettyS committed:
Refactor domain package

Shared package transfer rename to domain

Rename utils -> checkers.go.

Add Error to domain pkg.

Refactor functions to use domain.Error

Add TODOS refactor 'dom' prefix to domain
